tests/test_url_update_netloc.py

```diff
@@ -191,14 +191,32 @@ def test_with_port():
     assert str(url.with_port(8888)) == "http://example.com:8888"
 
 
+def test_with_port_normalization():
```

Comment:
It looks like this contains 3 distinct tests. Perhaps, it's reasonable to split them into 3 functions so they show up separately in the test output.